### PR TITLE
task: remove usage of resteasy-core-spi

### LINKED ISSUE #27242: Remove usage of resteasy-core-spi
### Description

As a follow-up to #26315 we should remove the usage of resteasy-core-spi as continued usage is likely to not be valid, and complete removal will help make migration issues manifest as class not found rather than subtle situations where something appears not to work.

### PR DESCRIPTION
**Pull request by shawkins**
closes: #27242

Things to review:
- usage of the test dependency for the resteasy classic implementation
- could rename KeycloakApplication and related classes to BaseXXX because they are now abstract.
- could give a more verbose name to ResteasyKeycloakApplication and related - could be ResteasyClassicKeycloakApplication instead
- ~~After #27368 - we can move all of the classes / singleton logic out of KeycloakApplication to clarify it's only applicable to resteasy classic.~~

### CODE REVIEW

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@shawkins Thank you for the changes.

Undertow seems to be broken, doesn't work in tests but more importantly doesn't work in the [dev server](https://github.com/keycloak/keycloak/blob/5b03ed174b546e5c16c6beb77376f074c9c5c4bf/docs/tests.md?plain=1#L28-L48). Could you please check it?

File: quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java [hunk 1]
```diff
@@ -637,9 +636,6 @@ AnnotationsTransformerBuildItem disableDefaultDataSourceHealthCheck() {
     void configureResteasy(CombinedIndexBuildItem index,
             BuildProducer<BuildTimeConditionBuildItem> buildTimeConditionBuildItemBuildProducer,
```
[the remark this answers is not in the document]
Comment (shawkins) [reply quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java:641]: It's still used https://github.com/keycloak/keycloak/pull/27387/files#diff-6ebe7f8db3e4dea9dbefd50eed883b8aac997e6c01127e283631e92e9d022835L647

**Comment by vmuzikar** [comment]
We'll also need a review from someone from @keycloak/core as this is touching their area.

File: services/src/test/java/org/keycloak/services/resteasy/ResteasyKeycloakApplication.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        classes.add(KeycloakMismatchedInputExceptionHandler.class);
+
+        singletons.add(new ObjectMapperResolver());
+        classes.add(WelcomeResource.class);
+
+        if (Profile.isFeatureEnabled(Profile.Feature.MULTI_SITE)) {
+            // If we are running in multi-site mode, we need to add a resource which to expose
+            // an endpoint for the load balancer to gather information whether this site should receive requests or not.
+            classes.add(LoadBalancerResource.class);
+        }
```
[the remark this answers is not in the document]
Comment (shawkins) [reply services/src/test/java/org/keycloak/services/resteasy/ResteasyKeycloakApplication.java:68]: I think so, it doesn't seem to be used by any of the other test logic.

**Comment by shawkins** [comment]
> @shawkins Thank you for the changes.
> 
> Undertow seems to be broken, doesn't work in tests but more importantly doesn't work in the [dev server](https://github.com/keycloak/keycloak/blob/5b03ed174b546e5c16c6beb77376f074c9c5c4bf/docs/tests.md?plain=1#L28-L48). Could you please check it?

Sure I can try it out manually.

Are we not running any undertow based tests in the ci build?

File: quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (vmuzikar) [reply quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java:641]: Ah, sorry. Added this comment yesterday when it was not used anymore. :)

**Comment by vmuzikar** [comment]
> Are we not running any undertow based tests in the ci build?

I'm afraid we aren't, although it still need to work mainly for dev purposes. :/ Something we should reconsider, TBH.

**Comment by shawkins** [comment]
> I'm afraid we aren't, although it still need to work mainly for dev purposes. :/ Something we should reconsider, TBH.

Ok, made that launch work - there was still one more reference to KeycloakApplication. The session factory was also missing an init call.

**Review by vmuzikar — APPROVED** [review]
LGTM, nice work @shawkins!

Last remaining thing is a review from @keycloak/core.

File: model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/CustomKeycloakTask.java [hunk 3]
```diff
@@ -73,18 +70,8 @@ public String getConfirmationMessage() {
     @Override
     public void setUp() throws SetupException {
         this.kcSession = ThreadLocalSessionContext.getCurrentSession();
-
         if (this.kcSession == null) {
-            // Probably running Liquibase from maven plugin. Try to create kcSession programmatically
```
Comment (pedroigor) [inline model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/CustomKeycloakTask.java:78]: Just curious, we were using this for what?

**Review by pedroigor — APPROVED** [review]
Nice

File: model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/CustomKeycloakTask.java (hunk 3 above)
Comment (shawkins) [reply model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/CustomKeycloakTask.java:78]: @pedroigor my guess would be there was a way to run migrations from a maven plugin. That doesn't seem to exist anymore.

**Review by jonkoops — APPROVED** [review]

**Pull request merged**